sky/resources.py

```diff
@@ -708,10 +709,29 @@ def _try_validate_local(self) -> None:
                         'Local/On-prem mode does not support custom '
                         'images.')
 
+    def extract_docker_image(self) -> Optional[str]:
+        if self.image_id is None:
+            return None
+        if len(self.image_id) == 1 and self.region in self.image_id:
+            image_id = self.image_id[self.region]
+            if image_id.startswith('docker:'):
+                return image_id[len('docker:'):]
+        return None
+
     def _try_validate_image_id(self) -> None:
         if self._image_id is None:
             return
 
+        if self.extract_docker_image() is not None:
+            # TODO(tian): validate the docker image exists / of reasonable size
+            if self.accelerators is not None:
+                for acc in self.accelerators.keys():
+                    if acc.lower().startswith('tpu'):
+                        with ux_utils.print_exception_no_traceback():
+                            raise ValueError(
+                                'Docker image is not supported for TPU VM.')
+            return
```

Comment:
Should we check if the cloud specified supports the docker? For example, `sky launch --cloud lambda --image-id docker:xxx` will fail to recognize that the cloud does not support docker at the moment?

Reply:
fixed now